pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -533,10 +515,17 @@ pub mod pallet {
 			let weight = Weight::from_parts(0, T::Message::max_encoded_len() as u64)
 				.saturating_add(LP_DEFENSIVE_WEIGHT);
 
-			match T::InboundMessageHandler::handle(domain_address, message) {
-				Ok(_) => (Ok(()), weight),
-				Err(e) => (Err(e), weight),
+			let mut count = 0;
+			for submessage in message.unpack() {
+				count += 1;
+
+				if let Err(e) = T::InboundMessageHandler::handle(domain_address.clone(), submessage)
+				{
+					return (Err(e), weight.saturating_mul(count));
+				}
 			}
+
+			(Ok(()), weight.saturating_mul(count))
```

Comment:
TODO. Debt to myself for the next PR. I think this could be an issue if we exceed the maximum during the batch processing. We should check that we can use all message weight before processing anything.

Reply:
Fixed in https://github.com/centrifuge/centrifuge-chain/pull/1951